geopandas/_vectorized.py

```diff
@@ -965,6 +1062,11 @@ def bounds(data):
 
 
 def transform(data, func):
+    if compat.SHAPELY_GE_20:
```

Comment:
Given the usefulness of vectorized `transform` beyond `to_crs`, any thoughts about exposing it on the GeoSeries or GeoDataFrame? 

We use the following pattern a lot and it would be awesome to have a single method:

```python
def func(x, y):
    …

gdf[“geometry”] = gdf[“geometry”].apply(lambda g: shapely.ops.transform(g, func))
```

Reply:
Also, looking at the shapely code, are you sure you need the get/set coordinates dance? Looks like you should just be able to call `return shapely.ops.transform(data, func, **kwargs)` directly.

Reply:
Something like.
```python
gdf.transform(func, inplace=True, **kwargs)
gdf1 = gdf.transform(func, inplace=False, **kwargs)
```

This would then make supporting multiple versions of shapely much easier for downstream while making it easy for users to get the speed ups from shapely 2 for those who have it.

Reply:
> Given the usefulness of vectorized `transform` beyond `to_crs`, any thoughts about exposing it on the GeoSeries or GeoDataFrame?

I think that can certainly be useful, but let's discuss that on a separate issue, since it's not strictly related to the rest of this PR

One challenge is that pandas already has a `transform` method, so the question is whether we want to override that or not (and for example expose this with a different name, such as `geom_transform`)

> Also, looking at the shapely code, are you sure you need the get/set coordinates dance? Looks like you should just be able to call `return shapely.ops.transform(data, func, **kwargs)` directly.

The reason we have this custom `transform` function in `_vectorized` (and which is only used inside `to_crs`) is because the signature of the function doesn't match exactly with `shapely.transform` (in shapely 2+), which takes a single 2D array, while here we pass x and y (and z) as separate arrays.

And for older shapely versions, we are actually using `shapely.ops.transform`. But this is a scalar function, and so for pygeos / shapely 2+, we can use a faster vectorized method (and we use `shapely.ops.transform` in a for loop to emulate that)

Reply:
There is some related discussion about this at https://github.com/geopandas/geopandas/issues/1175#issuecomment-544241270 (but didn't find a dedicated issue for this topic)